tests/avro/test_avro_producer.py

```diff
@@ -20,57 +20,66 @@
+    def test_produce_no_key(self):
+        value_schema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:9001'}, default_value_schema=value_schema)
         try:
-            producer.produce(topic='test', value={"name": 'abc"'}, value_schema=valueSchema, key='mykey')
-            self.fail("Should expect key_schema")
-        except Exception as e:
+            # No key should work
+            producer.produce(topic='test', value={"name": 'abc"'})
+        except SerializerError:
+            self.fail("Produce should accept empty keys")
+        except BaseException:
             pass
 
-    def test_produce_arguments(self):
-        value_schema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
-        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:0'}, default_value_schema=value_schema)
-
-        try:
+    def test_produce_no_value_schema(self):
+        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:9001'})
+        with self.assertRaises(SerializerError):
+            # Producer should not accept a value with no schema
             producer.produce(topic='test', value={"name": 'abc"'})
-        except Exception as e:
-            exc_type, exc_obj, exc_tb = sys.exc_info()
-            if exc_type.__name__ == 'SerializerError':
-                self.fail()
 
-    def test_produce_arguments_list(self):
-        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:0'})
+    def test_produce_no_key_schema(self):
+        value_schema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:9001'}, default_value_schema=value_schema)
         try:
-            producer.produce(topic='test', value={"name": 'abc"'}, key='mykey')
-        except Exception as e:
-            exc_type, exc_obj, exc_tb = sys.exc_info()
-            if exc_type.__name__ == 'SerializerError':
-                pass
+            producer.produce(topic='test', value={"name": 'abc"'}, key={"name": 'abc"'})
+        except SerializerError:
+            pass
+        except BaseException:
+            self.fail("If the key is provided as a dict an avro schema must also be provided")
 
+    def test_produce_value_and_key_schemas(self):
+        value_schema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:9001'}, default_value_schema=value_schema, default_key_schema=value_schema)
+        try:
+            producer.produce(topic='test', value={"name": 'abc"'}, key={"name": 'abc"'})
+        except SerializerError:
+            self.fail("Produce should accept both value and keys based on avro")
+        except BaseException:
+            pass
 
-def suite():
-    return unittest.TestLoader().loadTestsFromTestCase(TestAvroProducer)
+    def test_produce_string_key(self):
```

Comment:
My use case is extremely simple... I (sometimes) use primitive sting keys with Avro serialized values. I'm skipping Avro serialization for everything but `dict` and assuming the user knows what they are doing by by passing in exactly what they want for the key.

A better approach would be to add two additional parameters to the function, `serialize_key` and `serialize_value`, defaulting both to `True`. Is the acceptable? It would support both `str:avro`, `avro:str`, and `avro:avro` combinations.